src/core/thread/mle_router.cpp

```diff
@@ -4483,6 +4484,59 @@ Error MleRouter::SendTimeSync(void)
 }
 #endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
 
+Error MleRouter::DetachGracefully(otDetachGracefullyCallback aCallback, void *aContext)
+{
+    Error error = kErrorNone;
+
+    if (IsChild())
+    {
+        ExitNow(error = Mle::DetachGracefully(aCallback, aContext));
+    }
+
+    VerifyOrExit(!IsDetachingGracefully(), error = kErrorBusy);
+
+    OT_ASSERT(mDetachGracefullyCallback == nullptr);
+
+    mDetachGracefullyCallback = aCallback;
+    mDetachGracefullyContext  = aContext;
+
+    mDetachGracefullyTimer.Start(kDetachGracefullyTimeout);
+
+    if (IsRouter())
+    {
+        SendAddressRelease(&MleRouter::HandleDetachGracefullyAddressReleaseResponse, this);
+    }
+    else
+    {
+        mStopThreadTask.Post();
```

Comment:
How about here we schedule timer with 1 ms, like `mDetachGracefullyTimer.Start(1)` and then let the timer callback to call `Stop()`. 
I think with this we won't need to have a `mStopThreadTask` tasklet? and the callback is still indirectly invoked.

Reply:
mDetachGracefullyTimer.Start(0) may do the job.

Reply:
`mDetachGracefullyTimer.Start(0)` was the initial approach. @simonlingoogle suggested that using a tasklet is the conventional approach. I think we can remove the tasklet as the tasklet handler and the timer handler are identical for now.

Reply:
Done.